navigation/AppNavigator.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { createStackNavigator } from '@react-navigation/stack';
+import { Platform } from 'react-native';
+import { Ionicons } from '@expo/vector-icons';
+
+import Colors from '../theme/Colors';
+
+import DashboardScreen from '../screens/DashboardScreen';
+import SalesScreen, { SalesScreenNavOptions } from '../screens/SalesScreen';
+import SalesEntryScreen from '../screens/SalesEntryScreen';
+import CostsScreen, { CostsScreenNavOptions } from '../screens/CostsScreen';
+import CostsEntryScreen from '../screens/CostsEntryScreen';
+import SettingsScreen from '../screens/SettingsScreen';
+
+const defaultNavigationOptions = {
+  headerStyle: {
+    backgroundColor: Platform.OS === 'android' ? Colors.gray : 'white',
+  },
+  headerTintColor: Platform.OS === 'android' ? 'white' : Colors.gray,
+};
+
+const DashboardStackNavigator = createStackNavigator();
+
+const DashboardNavigator = () => {
+  return (
+    <DashboardStackNavigator.Navigator screenOptions={defaultNavigationOptions}>
+      <DashboardStackNavigator.Screen
+        name="Dashboard"
+        component={DashboardScreen}
+      />
+    </DashboardStackNavigator.Navigator>
+  );
+};
+
+const SalesStackNavigator = createStackNavigator();
+
+const SalesNavigator = () => {
+  return (
+    <SalesStackNavigator.Navigator screenOptions={defaultNavigationOptions}>
+      <SalesStackNavigator.Screen
+        name="SalesList"
+        component={SalesScreen}
+        options={SalesScreenNavOptions}
+      />
+      <SalesStackNavigator.Screen
+        name="SalesEntry"
+        component={SalesEntryScreen}
+      />
+    </SalesStackNavigator.Navigator>
+  );
+};
+
+const CostsStackNavigator = createStackNavigator();
+
+const CostsNavigator = () => {
+  return (
+    <CostsStackNavigator.Navigator screenOptions={defaultNavigationOptions}>
+      <CostsStackNavigator.Screen
+        name="CostsList"
+        component={CostsScreen}
+        options={CostsScreenNavOptions}
+      />
+      <CostsStackNavigator.Screen
+        name="CostsEntry"
+        component={CostsEntryScreen}
+      />
+    </CostsStackNavigator.Navigator>
+  );
+};
+
+const SettingsStackNavigator = createStackNavigator();
+
+const SettingsNavigator = () => {
+  return (
+    <SettingsStackNavigator.Navigator screenOptions={defaultNavigationOptions}>
+      <SettingsStackNavigator.Screen
+        name="Settings"
+        component={SettingsScreen}
+      />
+    </SettingsStackNavigator.Navigator>
+  );
+};
+
+const AppTabNavigator = createBottomTabNavigator();
+
+const AppNavigator = () => {
+  return (
+    <AppTabNavigator.Navigator
+      initialRouteName="Dashboard"
+      screenOptions={({ route }) => ({
+        tabBarIcon: ({ color, size }) => {
+          let iconName;
+
+          switch (route.name) {
```

Comment:
Actually it can be a simple object with screen name as a key:
```
const icons = {
  Dashboard: Platform.OS === 'android' ? 'md-stats' : 'ios-stats';
  // the rest
}
```

Reply:
good point :+1: I will refactor it